### PR TITLE
feat: add an option to be able to specify ssh key pair name, ebs volume size and type to the auto launch configuration

### LINKED ISSUE #83: Cannot specify SSH Key Pair Name in the Auto Lunch Configuration
There is a need to be able to login to the nodes to do debugging. This can be achieved by adding an SSH Key Pair in the Auto Lunch Configuration.

### LINKED ISSUE #84: Cannot specify the size and type of EBS in the Auto Launch Configuration
Size of EBS of is fixed to 100GB and type to gp2. Sometimes this is more than enough as we are using external resources anyway for storage like PVC for Stateful Sets and S3 and DynamoDB for logs and vault, respectively. Also if want to use standard or io1 it is not possible as of the moment.

### LINKED ISSUE #86: Cannot maximize Free Tier of DynamoDB
The Free Tier for DynamoDB is 25 Read Capacity Units and 25 Write Capacity Units. The default settings for both is 2 RCU and 2 WCU. Need to adjust this to maximize free tier.

I will create a separate PR for this.

### PR DESCRIPTION
**Pull request by joebertj**
There is a need to have the ability to be able to specify SSH Key Pair Name, EBS Volume Size and Type to the Auto Launch Configuration

fixes #83
fixes #84

### CODE REVIEW

File: modules/vault/main.tf [hunk 1]
```diff
@@ -43,9 +43,9 @@ resource "aws_s3_bucket" "vault-unseal-bucket" {
 // ----------------------------------------------------------------------------
 resource "aws_dynamodb_table" "vault-dynamodb-table" {
   name           = "vault-unseal-${var.cluster_name}-${local.vault_seed}"
-  billing_mode   = "PROVISIONED"
-  read_capacity  = 2
-  write_capacity = 2
+  billing_mode   = "PAY_PER_REQUEST"
```
Comment (ankitm123) [inline modules/vault/main.tf:46]: I would make them all variables instead of passing the values.

File: main.tf [hunk 2]
```diff
@@ -32,17 +32,24 @@ provider "template" {
...
+  max_node_count        = var.max_node_count
+  node_machine_type     = var.node_machine_type
+  spot_price            = var.spot_price
+  vpc_name              = var.vpc_name
+  vpc_subnets           = var.vpc_subnets
+  vpc_cidr_block        = var.vpc_cidr_block
+  force_destroy         = var.force_destroy
+  enable_spot_instances = var.enable_spot_instances
+  enable_key_name       = var.enable_key_name
+  key_name	        = var.key_name
```
Comment (ankitm123) [inline main.tf:49]: probably need to run `terraform fmt`

File: modules/vault/main.tf (hunk 1 above)
Comment (joebertj) [reply modules/vault/main.tf:46]: do we agree that we use Free Tier On-Demand 25 RCU/WCU as default and provide option to use Provision and specify RCU and WCU?

File: main.tf (hunk 2 above)
Comment (joebertj) [reply main.tf:49]: i tried to run it no effect. i am a noob on this rebase thing maybe i made a mistake somewhere.

**Comment by joebertj** [comment]
image
image

File: README.md [hunk 3]
```diff
@@ -138,7 +139,12 @@ The following sections provide a full list of configuration in- and output varia
 | node\_machine\_type | The instance type to use for the cluster's worker nodes | `string` | `"m5.large"` | no |
 | production\_letsencrypt | Flag to use the production environment of letsencrypt in the `jx-requirements.yml` file | `bool` | `false` | no |
 | region | The region to create the resources into | `string` | `"us-east-1"` | no |
-| spot_price | The ceiling price for spot instances | `string` | `"0.1"` | no |
+| spot\_price | The ceiling price for spot instances | `string` | `"0.1"` | no |
+| key\_name | The SSH Key Pair name | `string` | `""` | no |
+| volume\_type | The EBS Volume type | `string` | `"gp2"` | no |
+| volume\_size | The EBS Volume size in GB | `number` | `10` | no |
+| iops | The IOPS if chosen `volume_type` is `io1` | `number` | `0` | no |
+>>>>>>> feat: add an option to be able to specify ssh key pair name, ebs
```
Comment (hferentschik) [inline README.md:147]: this seems to be some left over from a git merge?

Comment (joebertj) [reply README.md]: fixed

**Comment by hferentschik** [comment]
I think here we also need to start with a rebase ;-)

**Comment by joebertj** [comment]
@hferentschik ok got it

**Comment by ankitm123** [comment]
@joebertj can you rebase, and run `make lint`? There are merge conflicts.

**Comment by joebertj** [comment]
@ankitm123 

yes will do. just got so busy. new work. new baby. we will implement jenkins x on my new company.

**Comment by ankitm123** [comment]
@joebertj jenkinsX 3 is going to be released soon, so probably better to look at it, instead of JX2.

**Pull request merged**